### PR TITLE
feat(graphene): Add span for grapqhl operation

### LINKED ISSUE #2765: GrapheneIntegration: Enhance Transaction Naming for Errors in Graphene GraphQL Operations
### Problem Statement

Currently when an exception or error occurs during the processing of a GraphQL operation by Graphene, the transaction name assigned to the reported issue is often not as descriptive as it could be. This lack of specificity in the transaction naming makes it challenging to quickly identify and address the source of the problem, and there are already existing workarounds such as described in [this article](https://jerrynsh.com/how-to-monitor-python-graphql-api-with-sentry/).

image

### Solution Brainstorm

Setting the transaction name of the current scope to the operation type and the operation name, would add more meaning to the issues, and one could see which operation was called at first glance.

image

This should be just a small change - i'm happy to do it 😄

### PR DESCRIPTION
**Pull request by czyber**
This commit adds a span for a graphql operation to the graphene integration

Fixes #2765

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

**Review by szokeasaurusrex — DISMISSED** [review]
Thank you for the contribution, and thanks for your patience as we work through our backlog of community PRs!

I noticed that your PR is still using our old, deprecated `Hub` APIs. We are going to need to replace these with the new corresponding `Scope` APIs before we can merge this PR. More information on how to do this is available in our [Migration Guide](https://docs.sentry.io/platforms/python/migration/1.x-to-2.x#custom-instrumentation). 

Is this a change you are comfortable with making yourself, or do you need us to take over the PR?

**Comment by antonpirker** [comment]
I take the freedom to remove the hubs from the pr

**Review by antonpirker — APPROVED** [review]
I think is is looking good now!

File: sentry_sdk/integrations/graphene.py [hunk 1]
```diff
@@ -17,7 +20,7 @@
 
 
 if TYPE_CHECKING:
-    from typing import Any, Dict, Union
+    from typing import Any, Dict, Generator, Union
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/graphene.py:23]:
```suggestion
    from collections.abc import Generator
    from typing import Any, Dict, Union
```

`typing.Generator` is a [deprecated alias](https://docs.python.org/3/library/typing.html#typing.Generator) to `collections.abc.Generator`

File: sentry_sdk/integrations/graphene.py [hunk 2]
```diff
@@ -106,3 +114,42 @@ def _event_processor(event, hint):
...
+    else:
+        _graphql_span = sentry_sdk.start_span(op=op, description=operation_name)
+
+    _graphql_span.set_data("graphql.document", source)
+    _graphql_span.set_data("graphql.operation.name", operation_name)
+    _graphql_span.set_data("graphql.operation.type", operation_type)
+
+    yield
+
+    _graphql_span.finish()
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/graphene.py:155]: We probably need to make sure we finish the span even in the case that we have an error in the `yield`
```suggestion
    try:
        yield
    finally:
         _graphql_span.finish()
```

**Comment by szokeasaurusrex** [comment]
Mostly looks good, see above

**Comment by antonpirker** [comment]
Committed your suggestions! Thanks a lot for the quick review!

**Pull request merged**